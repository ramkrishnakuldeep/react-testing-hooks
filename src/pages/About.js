import React from 'react'

function About() {
    return (
        <div>
            <h1 data-testid='about'>
                About
            </h1>
        </div>
    )
}

export default About
