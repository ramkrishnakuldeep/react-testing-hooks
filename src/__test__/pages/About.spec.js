import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import About from '../../pages/About';

afterEach(cleanup);

test('load About', async () => {
    const {asFragment} = render(<About />);
    expect(asFragment()).toMatchSnapshot()
});

test('About component test', () => {
    const {getByText, getByTestId} = render(<About />);
    expect(getByText('About')).toBeInTheDocument();
    expect(getByTestId('about')).toHaveTextContent('About');
})