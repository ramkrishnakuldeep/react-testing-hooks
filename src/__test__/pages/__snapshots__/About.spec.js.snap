// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load About 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="about"
    >
      About
    </h1>
  </div>
</DocumentFragment>
`;
